src/hotspot/cpu/riscv/vm_version_riscv.cpp

```diff
@@ -44,6 +44,11 @@ void VM_Version::initialize() {
     FLAG_SET_DEFAULT(AllocatePrefetchDistance, 0);
   }
 
+  if (!FLAG_IS_DEFAULT(CacheLineSize) && !is_power_of_2(CacheLineSize)) {
+    warning("CacheLineSize must be a power of 2");
```

Comment:
That was done to answer https://github.com/openjdk/jdk/pull/10718#discussion_r996410678. Ideally it would be detected at runtime. However, I don't know of any API in RISC-V or Linux to get that information.

Reply:
This could be an answer to the issue - https://github.com/riscv/riscv-profiles/blob/main/profiles.adoc#rva22-profiles 
Zic64b Cache blocks must be 64 bytes in size, naturally aligned in the address space.

The following mandatory feature was further restricted in RVA22U64:

Note | While the general RISC-V specifications are agnostic to cache block size, selecting a common cache block size simplifies the specification and use of the following cache-block extensions within the application processor profile. Software does not have to query a discovery mechanism and/or provide dynamic dispatch to the appropriate code. We choose 64 bytes at it is effectively an industry standard. Implementations may use longer cache blocks to reduce tag cost provided they use 64-byte sub-blocks to remain compatible. Implementations may use shorter cache blocks provided they sequence cache operations across the multiple cache blocks comprising a 64-byte block to remain compatible.

We can already create a flag: e.g. -XX:+UseRVA22U64BASE which will activate Zba, Zbb, Zicbom and set cache block size to 64-bytes

Reply:
I'll add a check like the following:
```
if (!UseZic64b) {
  if (FLAG_IS_DEFAULT(UseZicboz)) {
    FLAG_SET_DEFAULT(UseZicboz, false);
  }
}
```

Reply:
I've added https://github.com/openjdk/jdk/pull/10718/files#diff-7b173d6e5834de13749c8333192fef5a874628a67b90a5d8d06235d507542ac4R68-R79.